base/compiler/abstractinterpretation.jl

```diff
@@ -618,9 +615,16 @@ function abstract_call_method(interp::AbstractInterpreter, method::Method, @nosp
                 return MethodCallResult(Any, true, true, nothing, Effects())
             end
             add_remark!(interp, sv, washardlimit ? RECURSION_MSG_HARDLIMIT : RECURSION_MSG)
-            topmost = topmost::InferenceState
-            parentframe = topmost.parent
-            poison_callstack(sv, parentframe === nothing ? topmost : parentframe)
+            # TODO (#48913) implement a proper recursion handling for irinterp:
+            # This works just because currently the `:terminate` condition guarantees that
+            # irinterp doesn't fail into unresolved cycles, but it's not a good solution.
+            # We should revisit this once we have a better story for handling cycles in irinterp.
```

Comment:
I've added some comments to the code prefixed by `TODO (#48913)` to indicate where I've left some pending tasks. These tasks mostly involve implementing a proper recursion detection mechanism in irinterp. We can revisit these tasks when we allow irinterp on non-`:terminates` frames. The other notable one is to allow double inlining after irinterp resolves new `:calls`, which could potentially have a positive performance impact.